staging/src/k8s.io/apiserver/pkg/admission/plugin/resourcequota/resource_access.go

```diff
@@ -102,7 +110,43 @@ func (e *quotaAccessor) checkCache(quota *corev1.ResourceQuota) *corev1.Resource
 	return cachedQuota
 }
 
+const (
+	kcpByLogicalClusterAndNamespace                = "kcp-global-byLogicalClusterAndNamespace"
+	kcpClusterScopedQuotaNamespace                 = "admin"
+	kcpExperimentalClusterScopedQuotaAnnotationKey = "experimental.quota.kcp.dev/cluster-scoped"
+)
+
 func (e *quotaAccessor) GetQuotas(namespace string) ([]corev1.ResourceQuota, error) {
+	possibleClusterScopedQuotas, err := e.indexer.ByIndex(kcpByLogicalClusterAndNamespace, clusters.ToClusterAwareKey(e.clusterName, kcpClusterScopedQuotaNamespace))
+	if err != nil {
+		return nil, fmt.Errorf("error getting ResourceQuotas from namespace %q: %w", kcpClusterScopedQuotaNamespace, err)
+	}
+	// if there are no items held in our indexer, check our live-lookup LRU, if that misses, do the live lookup to prime it.
+	if len(possibleClusterScopedQuotas) == 0 {
+		lruItemObj, ok := e.liveLookupCache.Get(kcpClusterScopedQuotaNamespace)
+		if !ok || lruItemObj.(liveLookupEntry).expiry.Before(time.Now()) {
+			// TODO: If there are multiple operations at the same time and cache has just expired,
```

Comment:
This is an issue from 2016 - I didn't chase it, but I assume there's a fix. Might be useful to explain why we're not (or why we cannot) use it here, instead of linking to the issue, which made me thing it was still open.

Reply:
Huh, I see by expanding the diffs here that there's other TODOs like this already in this code.. ok.

Reply:
Yeah this was copy pasta